scripts/run.py

```diff
@@ -128,73 +132,100 @@ def plot_metrics(directory_name):
 
-    More information:
+    $ python run.py -c mle -b numpy -u https://www.hepdata.net/record/resource/1267798?view=true -m [750,100]
+    $ python run.py -c mle -u https://www.hepdata.net/record/resource/1267798?view=true -m [750,100]
+    $ python run.py -c mle -b numpy -p 1Lbb-likelihoods-hepdata -m [750,100]
+    $ python run.py -c interpolation -b jax -n 0 -mm fast
+    $ python run.py -c interpolation -b numpy -n 0 -mm slow
 
-      https://github.com/pyhf/pyhf-benchmark
 
-    """
+  More information:
 
-    if backend == "tensorflow":
-        tf.get_logger().setLevel("ERROR")
+    https://github.com/pyhf/pyhf-benchmark
 
+    """
     system = SystemStats()
     system.start()
 
-    pyhf.set_backend(backend)
-    print(f"Backend set to: {backend}")
+    if computation == "mle":
+        if backend == "tensorflow":
+            tf.get_logger().setLevel("ERROR")
+
+        pyhf.set_backend(backend)
+        print(f"Backend set to: {backend}")
+
+        if not model_point:
+            model_point_tuple = tuple()
+        else:
+            model_point_list = []
+            model_point = model_point[1:-1]
+            for item in model_point.split(","):
+                model_point_list.append(int(item))
+            model_point_tuple = tuple(model_point_list)
+
+        if url:
+            directory_name = downlaod(url)
+        elif path:
+            directory_name = open_local_file(path)
+        else:
+            print("Invalid command!")
+            print("Command help ....")
+            print(
+                "python run.py -b numpy -u https://www.hepdata.net/record/resource/1267798?view=true -m [750,100]"
+            )
+            print("python run.py -b numpy -p 1Lbb-likelihoods-hepdata -m [750,100]")
+            return
+
+        print(f"Dataset: {directory_name.name}")
+
+        background, signal_patch = get_bkg_and_signal(directory_name, model_point_tuple)
+
+        print("\nStarting fit\n")
+        fit_start_time = datetime.now()
+        CLs_obs, CLs_exp = calculate_CLs(background, signal_patch)
+        fit_end_time = datetime.now()
+        fit_time = fit_end_time - fit_start_time
+
+        print(f"fit {directory_name.name} in {fit_time} seconds\n")
+        print(f"CLs_obs: {CLs_obs}")
+        print(f"CLs_exp: {CLs_exp}")
+
+        if url:
+            delete_downloaded_file(directory_name)
+    elif computation == "interpolation":
+        if mode not in ["slow", "fast"]:
+            raise ValueError(f"The mode must be either 'slow' or 'fast', not {mode}.")
+
+        if number != "4p" and not (number.isdigit() and 0 <= int(number) <= 4):
+            raise ValueError(
+                f"The num must be integer in range [0, 4] or 4p, not {number}."
+            )
+
+        h, a = random_histosets_alphasets_pair()
+
+        class_name = (
+            "_slow_code" + str(number) if mode == "slow" else "code" + str(number)
+        )
+        interpolator = getattr(pyhf.interpolators, class_name)
```

Comment:
For this and the "number" code above I'd just remove it and get the interpolator object from `pyhf`'s `get` API: https://github.com/scikit-hep/pyhf/blob/d35f4b487d3a03d42edb6f3a49122283fedd5d59/src/pyhf/interpolators/__init__.py#L30